charmcraft/linters.py

```diff
@@ -202,12 +202,20 @@ def run(self, basedir: pathlib.Path) -> str:
 ]
 
 
-def analyze(config: config.Config) -> List[CheckResult]:
+def analyze(config: config.Config, basedir: pathlib.Path) -> List[CheckResult]:
     """Run all checkers and linters."""
     all_results = []
     for checker_class in CHECKERS:
+        # do not run the ignored ones
```

Comment:
I personally prefer reductions here, also, maybe we should have CHECKERS split in two groups in the future, could be done at module load time

Reply:
On one hand, `CHECKERS` should stay as a single list, as in the future attributes and linters may share state (analysis/explorations done by one of those that are useful to others, so will be stored in the shared_state to not repeat it).

We could split it in two groups at module load time, but you will be running that *everytime* the process starts (as the config imports linters), and it will be waste if the Builder is not used...